lein repl (2.8.2) throws an exception on Ubuntu 18.04 and 16.04

I get the following stack trace when running `lein repl` on Ubuntu 18.04 and 16.04 (Leiningen 2.8.2 on Java 1.8.0_191 OpenJDK 64-Bit Server VM):

    Exception in thread "Thread-1" java.lang.AssertionError: Assert failed: transport-fn
	at nrepl.core$connect.invokeStatic(core.clj:168)
	at nrepl.core$connect.doInvoke(core.clj:168)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at nrepl.ack$send_ack.invokeStatic(ack.clj:45)
	at nrepl.ack$send_ack.invoke(ack.clj:41)
	at nrepl.server$start_server.invokeStatic(server.clj:146)
	at nrepl.server$start_server.doInvoke(server.clj:105)
	at clojure.lang.RestFn.invoke(RestFn.java:1096)
	at user$eval747.invokeStatic(NO_SOURCE_FILE:0)
	at user$eval747.invoke(NO_SOURCE_FILE)
	at clojure.lang.Compiler.eval(Compiler.java:6927)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.eval(Compiler.java:6890)
	at clojure.core$eval.invokeStatic(core.clj:3105)
	at clojure.core$eval.invoke(core.clj:3101)
	at leiningen.core.eval$fn__4211.invokeStatic(eval.clj:343)
	at leiningen.core.eval$fn__4211.invoke(eval.clj:333)
	at clojure.lang.MultiFn.invoke(MultiFn.java:233)
	at leiningen.core.eval$eval_in_project.invokeStatic(eval.clj:367)
	at leiningen.core.eval$eval_in_project.invoke(eval.clj:357)
	at leiningen.repl$server$fn__5985.invoke(repl.clj:276)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:646)
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1881)
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1881)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:650)
	at clojure.core$bound_fn_STAR_$fn__4671.doInvoke(core.clj:1911)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.lang.Thread.run(Thread.java:748)
    REPL server launch timed out.

Leiningen 2.8.1 runs without error.

--- a/leiningen-core/src/leiningen/core/project.clj
+++ b/leiningen-core/src/leiningen/core/project.clj
@@ -554,7 +554,7 @@
                 :test-selectors {:default (with-meta '(constantly true)
                                             {:displace true})}
                 ;; bump deps in leiningen's own project.clj with these
-                :dependencies '[^:displace [nrepl/nrepl "0.5.2"
+                :dependencies '[^:displace [nrepl/nrepl "0.5.3"
                                             :exclusions [org.clojure/clojure]]
                                 ^:displace [clojure-complete "0.2.5"
                                             :exclusions [org.clojure/clojure]]]

--- a/leiningen-core/test/leiningen/core/test/project.clj
+++ b/leiningen-core/test/leiningen/core/test/project.clj
@@ -44,7 +44,7 @@
                                [stencil/stencil "0.2.0"]
                                [~(symbol "net.3scale" "3scale-api") "3.0.2"]
                                [clj-http/clj-http "3.4.1"]
-                               [nrepl/nrepl "0.5.2"
+                               [nrepl/nrepl "0.5.3"
                                 :exclusions [[org.clojure/clojure]]]
                                [clojure-complete/clojure-complete "0.2.5"
                                 :exclusions [[org.clojure/clojure]]]],

--- a/project.clj
+++ b/project.clj
@@ -17,7 +17,7 @@
                  ;; needed for uberjar
                  [commons-lang "2.6"]
                  ;; needed for repl
-                 [nrepl "0.5.2"]
+                 [nrepl "0.5.3"]
                  ;; needed for change
                  [net.cgrand/sjacket "0.1.1" :exclusions [org.clojure/clojure]]
                  ;; bump versions of various common transitive deps

--- a/resources/leiningen/bootclasspath-deps.clj
+++ b/resources/leiningen/bootclasspath-deps.clj
@@ -33,7 +33,7 @@
  net.cgrand/parsley "0.9.3"
  net.cgrand/regex "1.1.0"
  net.cgrand/sjacket "0.1.1"
- nrepl "0.5.2"
+ nrepl "0.5.3"
  org.apache.commons/commons-lang3 "3.5"
  org.apache.httpcomponents/httpclient "4.5.3"
  org.apache.httpcomponents/httpcore "4.4.6"

